sources/api/sundog/src/main.rs

```diff
@@ -245,8 +244,18 @@ where
             key_type: KeyType::Data,
             key: &setting_str,
         })?;
+
         // Don't clobber settings that are already populated
-        if populated_settings.contains(&setting) {
+        // We're checking for prefix matches here because some generators are for settings that map
+        // to a top level struct that might have subfields. For example, a settings generator for
+        // `settings.boot` encompasses both `settings.boot.kernel` and `settings.boot.init`.
```

Comment:
> We could use `prairiedog` as a generator for `settings.boot.kernel-parameters` and `settings.boot.init-parameters`, but that would result in two program invocations rather than just one, and we'd essentially have to do the same work twice, throwing away half of it each time.

If we're talking about before this change, that would be correct. We'd have to specify `metadata.settings.boot.kernel-parameters.setting-generator` and `metadata.settings.boot.init-parameters.setting-generator` separately.

But with the prefix matching, we can specify settings generator for a top-level setting like `settings.boot` and `prairiedog` would just be invoked once to return a serialized json string that contains everything under `settings.boot`. Another neat thing is if either `kernel-parameters` or `init-parameters` is already set, then sundog would skip over the settings generation for the top-level setting, i.e. `settings.boot`.